### PR TITLE
fix dlrm accuracy test in max-autotune

### PR DESCRIPTION
**Pull request by shunting314**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #122012

torchrec_dlrm training fail the accuracy check when max-autotune is enabled.

I found there is no real issue in PT2. We fail to get fp64 reference results for the accuracy check. In max-autotune mode numerical may change a bit and cause the cosine similarity check fail. Using fp64 baseline is more reliable and make the test pass.

The reason why we are not using a fp64 baseline earlier is because torchrec uses a dataclass [Batch](https://github.com/pytorch/torchrec/blob/99e6e669b5a6cc73f6405d654f81427e19fc81b4/torchrec/datasets/utils.py#L28) to represent the input. We use pytree to cast model and inputs to fp64. pytree can not look into a dataclass. My fix is to convert the dataclass to namedtuple to be more pytree friendly

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang

### CODE REVIEW

**Review by jansel — APPROVED** [review]

**Review by eellison — APPROVED** [review]

**Comment by shunting314** [comment]
The failure regarding pytorch_unet is unrelated.

The failure for torchrec_dlrm in dynamic_shape case is interesting. Actually previously torchrec_dlrm can not run because the benchmark script can not detect the batch dimension for dynamic shape (due to the same reason fixed by this PR). Somehow the `benchmarks/dynamo/check_accuracy.py` script does not capture this kind of failures so it shows as pass in CI. With the fix in this PR, the model fail with some error regarding dynamic shape and this error is captured by CI. I'll skip this model for dynamic shape training. We probably should enhance `benchmarks/dynamo/check_accuracy.py`
